fairscale/nn/data_parallel/fully_sharded_data_parallel.py

```diff
@@ -1069,6 +1069,53 @@ def assert_state(self, state: TrainingState) -> None:
             self.training_state == state
         ), f"expected to be in state {state} but current state is {self.training_state}"
 
+    @staticmethod
+    @contextlib.contextmanager
+    def enable_auto_wrap(**kwargs) -> Iterator[None]:
+        """
+        Context manager to wrap modules in FullyShardedDataParallel.
+        Useful for when you'd like to apply the same parameters to all child modules that you wrap.
+
+            ...
+            with FullyShardedDataParallel.enable_auto_wrap(**fully_sharded_params):
+                self.l1 = FullyShardedDataParallel.auto_wrap(torch.nn.Linear(5, 5))
+                self.l2 = FullyShardedDataParallel.auto_wrap(torch.nn.Linear(5, 5))
+
+        """
+        with FullyShardedAutoWrap(**kwargs):
+            yield
+
+    @staticmethod
+    def auto_wrap(module, **kwargs):
+        if FullyShardedAutoWrap.in_autowrap_context:
+            kwargs = {**FullyShardedAutoWrap.kwargs, **kwargs}
+            return FullyShardedDataParallel(module, **kwargs)
+        return module
+
+
+class FullyShardedAutoWrap:
```

Comment:
I think some docstring might be needed, which might be the reason CI is failing. I use this alias to do auto-format to make CI happy:

```
alias fs_test='black . && isort . && flake8 && mypy --ignore-missing-imports --scripts-are-modules --pretty . && echo ALL GOOD'
```